src/ElasticSearchIndexer.php

```diff
@@ -110,6 +110,10 @@ public function buildIndex(Iterator $documents): int
             if ($blockSize === self::INDEXING_BLOCK_SIZE) {
                 $clientResponse = $this->client->bulk($params);
 
+                if ($clientResponse['errors'] === true) {
+                    throw new ClientErrorResponseException($this->parseErrors($clientResponse));
+                }
```

Comment:
You can create a private method that does it to avoid repetition:

private function validateErrors(array $clientResponse): void
{
    if ($clientResponse['errors'] === true) {
        throw new ClientErrorResponseException($this->parseErrors($clientResponse));
    }
}

Reply:
I would like to keep this in method body as it is easier to read when you see directly in a method that it is throwing an error when response hs error flag enabled.